src/ensembl/src/content/app/custom-download/containers/content/result-holder/ResultHolder.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { RootState } from 'src/store';
+import {
+  getPreviewResult,
+  getIsLoadingResult
+} from '../../../state/customDownloadSelectors';
+
+import { getAttributes } from '../attributes-accordion/state/attributesAccordionSelector';
+import { getFilters } from '../filter-accordion/state/filterAccordionSelector';
+
+import AttributesSection from 'src/content/app/custom-download/types/Attributes';
+
+import {
+  setPreviewResult,
+  setIsLoadingResult,
+  fetchPreviewResult
+} from '../../../state/customDownloadActions';
+
+import styles from './ResultHolder.scss';
+
+import { CircleLoader } from 'src/shared/loader/Loader';
+
+import {
+  getSelectedAttributes,
+  formatResults,
+  getEndpointUrl
+} from './helpers';
+
+type Props = StateProps & DispatchProps;
+
+const ResultHolder = (props: Props) => {
+  const selectedAttributes: AttributesSection = getSelectedAttributes(
+    props.attributes
+  );
+
+  useEffect(() => {
+    if (!selectedAttributes.length && props.preview.results) {
+      props.clearPreviewResult([]);
```

Comment:
`clearPreviewResult` is defined below as `setPreviewResult.success`. And in `customDownloadActions.ts` `setPreviewResult.success` is described as a function that takes an argument of  `{ preview: {} }`, if I am reading that correctly. How does it agree with this clearPreviewResult that takes an empty array?

Reply:
I've updated the above line to pass {} instead of [] and changed `preview: {}` to `preview: any` for the time being as I do not have a proper structure for it.